flocker/node/_loop.py

```diff
@@ -379,18 +390,26 @@ def got_local_state(state_changes):
 
             # XXX And for this update to be the side-effect of an output
             # resulting.
-            self._maybe_send_state_to_control_service(state_changes)
+            sent_state = self._maybe_send_state_to_control_service(
+                state_changes)
 
             action = self.deployer.calculate_changes(
                 self.configuration, self.cluster_state
             )
             LOG_CALCULATED_ACTIONS(calculated_actions=action).write(
                 self.fsm.logger)
-            return run_state_change(action, self.deployer)
+            ran_state_change = run_state_change(action, self.deployer)
+            DeferredContext(ran_state_change).addErrback(
+                writeFailure, self.fsm.logger)
```

Comment:
I think it would be beneficial to supply a system here.  `DeferredContext` makes sure `writeFailure` is run in the context of the ... `LOG_CONVERGE` action (I think?  at the very least, whatever action is on active here) but it won't associate the failure with the `run_state_change` operation.  I think making this association is the purpose of passing a system to `writeFailure`.